kore/src/Kore/Syntax/Not.hs

```diff
@@ -0,0 +1,52 @@
+{-|
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+
+-}
+
+{-# LANGUAGE TemplateHaskell #-}
+
+module Kore.Syntax.Not
+    ( Not (..)
+    ) where
+
+import           Control.DeepSeq
+                 ( NFData (..) )
+import qualified Data.Deriving as Deriving
+import           Data.Hashable
+import qualified Data.Text.Prettyprint.Doc as Pretty
+import           GHC.Generics
+                 ( Generic )
+
+import Kore.Sort
+import Kore.Unparser
+
+{-|'Not' corresponds to the @\not@ branches of the @object-pattern@ and
+@meta-pattern@ syntactic categories from the Semantics of K,
+Section 9.1.4 (Patterns).
+
+'notSort' is both the sort of the operand and the sort of the result.
+
+-}
+data Not sort child = Not
```

Comment:
It's not obvious why the `sort` argument is needed. Is it to allow the Applicative stuff?